pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -687,5 +687,5 @@ func decreaseAndCheckMaxUnavailable(maxUnavailable *int) bool {
 	}
 	val := *maxUnavailable - 1
 	*maxUnavailable = val
-	return val <= 0
+	return val < 0
```

Comment:
There should be no further adjustments here, and the original writing method should be maintained.